packages/api/src/router/activity.ts

```diff
@@ -0,0 +1,95 @@
+import { z } from "zod";
+
+import { adminProcedure, createTRPCRouter } from "../trpc";
+
+export const activityRouter = createTRPCRouter({
+  getActivity: adminProcedure
+    .input(z.object({ id: z.number().nonnegative().optional() }))
+    .query(async ({ ctx, input }) => {
+      return input.id
+        ? await ctx.db.activity.findFirst({
+            where: { id: input.id },
+          })
+        : await ctx.db.activity.findMany({
+            orderBy: { day: "desc" },
+          });
+    }),
+  getSchedule: adminProcedure
+    .input(z.object({ day: z.string() }))
+    .query(async ({ ctx, input }) => {
+      return await ctx.db.activity.findMany({
+        where: {
+          day: {
+            gte: new Date(input.day),
+            lte: new Date(
+              new Date(input.day).getTime() + 7 * 24 * 60 * 60 * 1000,
+            ),
+          },
+        },
+        orderBy: { day: "desc" },
+      });
+    }),
+  createActivity: adminProcedure
+    .input(
+      z.object({
+        name: z.string(),
+        day: z.string(), // format YYYY-MM-DD
+        startTime: z.string(), // format HH:MM
+        endTime: z.string(), // format HH:MM
+        leader: z.string(),
+        location: z.string(),
+      }),
+    )
+    .mutation(async ({ ctx, input }) => {
+      await ctx.db.activity.create({
+        data: {
+          name: input.name,
+          day: new Date(input.day),
+          startTime: new Date("1970/01/01 " + input.startTime + "Z"),
+          endTime: new Date("1970/01/01 " + input.endTime + "Z"),
+          leader: input.leader,
+          location: input.location,
+        },
```

Comment:
Should we be returning the new activity when an activity is successfully created?

Reply:
Fixed thank you for catching!